eng/tools/openapi-diff-runner/src/utils/common-utils.ts

```diff
@@ -111,13 +112,11 @@ export function getVersionFromInputFile(filePath: string, withPreview = false):
     }
   }
 
-  // If no regex match found, return the immediate folder name (parent directory of the file)
-  if (segments && segments.length > 1) {
-    // Get the folder name that contains the file (last segment before filename)
-    const folderName = segments[segments.length - 2];
-    if (folderName) {
-      return folderName;
-    }
+  // If no regex match found, try to read version from file content
+  if (existsSync(filePath)) {
+    const fileContent = readFileSync(filePath, "utf8");
+    const parsedContent = JSON.parse(fileContent);
+    return parsedContent?.info?.version || "";
   }
 
   return "";
```

Comment:
Does it ever make sense to return "" from getVersionFromInputFile()?  Should we throw instead?